### PR TITLE
Implement trip form validation

### PR DESCRIPTION
**Pull request by tanruiquan**
The form validation design is inspired by [this](https://www.youtube.com/watch?v=7oMlz-pMSAI) youtube video.

### CODE REVIEW

**Review by keanecjy — APPROVED** [review]
LGTM. 👍

File: TourMate/TourMate/Frontend/Trip/Views/EditTripView/EditTripView.swift [hunk 1]
```diff
@@ -27,11 +27,13 @@ struct EditTripView: View {
                         DatePicker(
                             "Start Date",
                             selection: $viewModel.trip.startDate,
+                            in: Date()...,
                             displayedComponents: [.date]
                         )
                         DatePicker(
                             "End Date",
                             selection: $viewModel.trip.endDate,
+                            in: viewModel.fromStartDate,
```
Comment (keanecjy) [inline TourMate/TourMate/Frontend/Trip/Views/EditTripView/EditTripView.swift:36]: We also need to enforce that end date > start date as well

Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Trip/Views/EditTripView/EditTripView.swift:36]: User won't be able to choose a end date that is < start date

Comment (tanruiquan) [reply TourMate/TourMate/Frontend/Trip/Views/EditTripView/EditTripView.swift:36]: User won't be able to choose a end date that is < start date

Comment (keanecjy) [reply TourMate/TourMate/Frontend/Trip/Views/EditTripView/EditTripView.swift:36]: Okay my bad, didn't realize

**Pull request merged**